com.unity.render-pipelines.universal/Tests/Runtime/Light2DTests.cs

```diff
@@ -140,5 +161,247 @@ public void CachedMeshDataIsOverriddenByRuntimeChanges()
             Assert.AreNotEqual(vertexCount, light.lightMesh.triangles.Length);
             Assert.AreNotEqual(triangleCount, light.lightMesh.vertices.Length);
         }
+        
+        [UnityTest]
+        public IEnumerator OnDisableRendererDataBatching_DisablesBatching()
+        {
+            var asset = GraphicsSettings.renderPipelineAsset as UniversalRenderPipelineAsset;
+            var rendererData = asset.scriptableRendererData as Renderer2DData;
+            
+            var shapePath = new Vector3[4] { new Vector3( 0, 0, 0), new Vector3(1,0,0), new Vector3(1, 1, 0), new Vector3(0, 1, 0) };
+            var light1 = m_TestObjectBatched1.AddComponent<Light2D>();
+            var light2 = m_TestObjectBatched2.AddComponent<Light2D>();
+            
+            rendererData.enableBatching = false;
+            light1.lightType = Light2D.LightType.Freeform;
+            light1.SetShapePath(shapePath);
+            light1.UpdateMesh(true);
+            light2.lightType = Light2D.LightType.Freeform;
+            light2.SetShapePath(shapePath);
+            light2.UpdateMesh(true);
+            
+            m_Camera.GetComponent<Camera>().Render();
+            yield return null;            
+
+            Assert.AreEqual(0, Light2DBatch.s_Batches);
+        }
+        
+        [UnityTest]
+        public IEnumerator OnEnableRendererDataBatching_EnablesBatching_AdditiveStyle()
+        {
+            var asset = GraphicsSettings.renderPipelineAsset as UniversalRenderPipelineAsset;
+            var rendererData = asset.scriptableRendererData as Renderer2DData;
+            
+            var shapePath = new Vector3[4] { new Vector3( 0, 0, 0), new Vector3(1,0,0), new Vector3(1, 1, 0), new Vector3(0, 1, 0) };
+            var light1 = m_TestObjectBatched1.AddComponent<Light2D>();
+            var light2 = m_TestObjectBatched2.AddComponent<Light2D>();
+            
+            rendererData.enableBatching = true;
+            light1.lightType = Light2D.LightType.Freeform;
+            light1.SetShapePath(shapePath);
+            light1.UpdateMesh(true);
+            light2.lightType = Light2D.LightType.Freeform;
+            light2.SetShapePath(shapePath);
+            light2.UpdateMesh(true);
+            
+            Light2DBlendStyle style = m_Data.lightBlendStyles[0];
+            style.blendMode = Light2DBlendStyle.BlendMode.Additive;
+            m_Data.SetLightBlendStyle(style, 0);            
+
+            m_Camera.GetComponent<Camera>().Render();
+            yield return null;            
+            
+            Assert.AreEqual(1, Light2DBatch.s_Batches);
+        }     
+        
+        [UnityTest]
+        public IEnumerator OnEnableRendererDataBatching_DisablesBatching_MultiplicativeStyle()
```

Comment:
No longer valid.

Reply:
Fixed in c758a3b9ffc184b30ba26bef8fd0162b65f4725d